src/main/java/handler/BalanceOperationHandler.java

```diff
@@ -0,0 +1,17 @@
+package handler;
+
+import java.util.Map;
+import model.FruitTransaction;
+
+public class BalanceOperationHandler implements OperationHandler {
+    private final Map<String, Integer> storage;
+
+    public BalanceOperationHandler(Map<String, Integer> storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        storage.put(transaction.getFruit(), transaction.getQuantity());
```

Comment:
Be cautious with directly updating the storage. It's a good practice to first validate the input data. In this case, ensure that the quantity is not negative before updating the storage.